mmdet/models/detectors/cascade_rcnn.py

```diff
@@ -135,13 +134,25 @@ def forward_train(self,
             lw = self.train_cfg.stage_loss_weights[i]
 
             # assign gts and sample proposals
-            assign_results, sampling_results = multi_apply(
-                assign_and_sample,
-                proposal_list,
-                gt_bboxes,
-                gt_bboxes_ignore,
-                gt_labels,
-                cfg=rcnn_train_cfg)
+            sampling_results = []
+            if self.with_bbox or self.with_mask:
+                bbox_assigner = build_assigner(rcnn_train_cfg.assigner)
+                bbox_sampler = build_sampler(
+                    rcnn_train_cfg.sampler, context=self, stages=i)
```

Comment:
I suggest that we add a stage indicator like `self.current_stage = i`, thus it can be obtained in the sampler via `context.current_stage` instead of passing an additional argument here.